### PR TITLE
Boilerplate backend

### PR DESCRIPTION
**Pull request by PedroMagalhaes20**
Backend for the Boilerplate

### CODE REVIEW

**Comment by vitorlofonseca** [comment]
You need to modify no other file than the controller. Please, rebase your branch properly

**Review by vitorlofonseca — APPROVED** [review]

File: backend/src/main/java/saci/application/HelloWorldController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+public class HelloWorldController {
+
+    @GetMapping
+    public String hello() {
+        return "Hello world";
+    }
+
+    @PostMapping
+    public ResponseEntity<String> helloPost(@RequestBody String requestBody) {
+        // Process the POST request
```
Comment (vascolusitan) [inline backend/src/main/java/saci/application/HelloWorldController.java:21]: Avoid ChatGPT coments. :)

File: backend/src/main/java/saci/configuration/WebMvcConfiguration.java [hunk 2]
```diff
@@ -9,15 +9,24 @@
...
 
     @Bean
     public WebMvcConfigurer corsConfigurer() {
         return new WebMvcConfigurer() {
             @Override
             public void addCorsMappings(CorsRegistry registry) {
-                registry.addMapping("/**").allowedMethods("*").allowedOrigins(corsUrl);
+                // Client CORS configuration: Allow GET and POST requests
+                registry.addMapping("/**")
+                        .allowedMethods("GET", "POST")
```
Comment (vascolusitan) [inline backend/src/main/java/saci/configuration/WebMvcConfiguration.java:25]: Does the client need POST type requests?

File: backend/src/main/java/saci/configuration/WebMvcConfiguration.java [hunk 3]
```diff
@@ -9,15 +9,24 @@
...
+    @Value("${CORS_URL_ADMIN:http://localhost:8082}")
+    private String corsUrlAdmin;
 
     @Bean
     public WebMvcConfigurer corsConfigurer() {
         return new WebMvcConfigurer() {
             @Override
             public void addCorsMappings(CorsRegistry registry) {
-                registry.addMapping("/**").allowedMethods("*").allowedOrigins(corsUrl);
+                // Client CORS configuration: Allow GET and POST requests
```
Comment (vascolusitan) [inline backend/src/main/java/saci/configuration/WebMvcConfiguration.java:23]: Avoid comments.

File: backend/src/main/java/saci/configuration/WebMvcConfiguration.java [hunk 4]
```diff
@@ -9,15 +9,24 @@
...
+                registry.addMapping("/**")
+                        .allowedMethods("GET", "POST")
+                        .allowedOrigins(corsUrlClient);
+
+                // Admin CORS configuration: Allow all methods
```
Comment (vascolusitan) [inline backend/src/main/java/saci/configuration/WebMvcConfiguration.java:28]: Avoid comments.

File: backend/src/main/resources/application.yml [hunk 5]
```diff
@@ -9,3 +9,5 @@ spring:
   flyway:
     enabled: true
     repair: true
+
```
Comment (vascolusitan) [inline backend/src/main/resources/application.yml:12]: Check your file changes before commiting.

File: backend/src/main/java/saci/configuration/WebMvcConfiguration.java (hunk 2 above)
Comment (PedroMagalhaes20) [reply backend/src/main/java/saci/configuration/WebMvcConfiguration.java:25]: Victor requested me to add a post

**Pull request merged**